lib/raven/integrations/sidekiq.rb

```diff
@@ -5,9 +5,14 @@ module Raven
   class SidekiqCleanupMiddleware
     def call(_worker, _job, _queue)
       yield
+    rescue Exception
+      exception_raised = true
+      raise
     ensure
-      Context.clear!
-      BreadcrumbBuffer.clear!
+      unless exception_raised
```

Comment:
I added a comment to explain this: https://github.com/getsentry/raven-ruby/pull/657/commits/72544b7ef73ed319e39ce9bd0127894ebe8b2984#diff-f18bf682cd7bdb47e4e6bdd40c800adcR12